File: 231231/순 간 이 동/teleportation.cpp
#include <iostream>
#include <climits>
#include <cstdlib>

using namespace std;
int a, b, x, y;
int ans = INT_MAX;

int main() {
    cin >> a >> b >> x >> y;
    // case 1. a->b로 바로 가기
    ans = min(ans, abs(b-a));

    // case 2. a-> x -> y -> b
    ans = min(ans, abs(a-x) + abs(y-b));

    // case 3. a -> y -> x -> b
    ans = min(ans, abs(a-y) + abs(x-b));
    cout << ans;
    return 0;
}

File: 240131/순 간 이 동/teleportation.py
import sys
INT_MAX = sys.maxsize

# 변수 선언 및 입력
a, b, x, y = tuple(map(int, input().split()))

ans = INT_MAX

# 1. a->b
ans = min(ans, abs(b-a))

# 2. a->x->y->b
ans = min(ans, abs(x-a) + abs(b-y))

# 3. a->y->x->b
ans = min(ans, abs(y-a) + abs(b-x))

print(ans)